src/index.ts

```diff
@@ -1,26 +1,63 @@
-import { getTargetOrigin, TestBoxConfig } from "./config";
-import { info } from "./utils/logging";
+import type { TestBoxConfig, LoginHandler } from "./config";
+import { sendMessageToTestBox } from "./messaging";
 import {
-  isValidIncomingTestBoxMessage,
-  sendMessageToTestBox,
-} from "./messaging";
-import { routeMessage } from "./router";
-import { INITIALIZE_REQUEST } from "./messaging/outgoing";
+  INITIALIZE_REQUEST,
+  LOGIN_HANDLER_REGISTERED,
+} from "./messaging/outgoing";
 import { LoginEvent, NavigateEvent } from "./messaging/incoming";
+import { messageEventCallback } from "./message-event";
+import { routeMessage } from "./router";
 
 let tbxStarted = false;
+let isLoginHandlerRegistered = false;
+
+export let navigateHandler: (data: NavigateEvent) => void = (data) => {
+  window.location.href = data.url;
+};
+export let loginHandler: (data: LoginEvent) => Promise<string | boolean> =
+  undefined;
+function eventCallbackMiddleware(ev: MessageEvent<any>) {
+  messageEventCallback(ev, { navigateHandler, loginHandler });
+}
+
+export async function registerLoginHandler(newLoginHandler: LoginHandler) {
+  if (isLoginHandlerRegistered) {
+    throw new Error("LoginHandler already registered!");
+  }
 
-export let navigateHandler: (data: NavigateEvent) => void;
-export let loginHandler: (
-  data: LoginEvent
-) => Promise<string | boolean> = async () => false;
+  if (!tbxStarted) {
+    throw new Error(
+      "StartTestbox function must be called before registering login handler!"
+    );
+  }
+
+  window.removeEventListener("message", eventCallbackMiddleware);
+
+  loginHandler = newLoginHandler;
+
+  if (window?.__loginMessagesQueue && window.__loginMessagesQueue.length > 0) {
+    for (const loginMessage of window.__loginMessagesQueue) {
+      routeMessage(loginMessage, {
+        navigate: navigateHandler,
+        login: loginHandler,
+      });
+    }
+    window.__loginMessagesQueue = [];
+  }
+
+  window.addEventListener("message", eventCallbackMiddleware);
+
+  sendMessageToTestBox(LOGIN_HANDLER_REGISTERED);
+  isLoginHandlerRegistered = true;
+}
 
 export function startTestBox(config?: TestBoxConfig) {
   if (tbxStarted) {
     return;
   }
 
   window.__tbxConfig = config;
+  window.__loginMessagesQueue = [];
```

Comment:
```suggestion
  // To be registered by registerLoginHandler in case the login event arrives before the handler is registered.
  window.__tbxLoginEvent = null;
```

Two things here
- worth prefixing what we register into the window with `tbx` so its clear for people looking into the window object that thats something created by testbox
- thinking through rn, there is no scenario where multiple login messages would be reasonable, we should not get two, and even if we do get two, only one should be ran, so the last can overwrite anything that came before

Reply:
What I mean with the 2nd item is that we dont need this to be a list, but rather, just a single item